modules/locking/src/main/java/io/uhndata/cards/locking/api/LockManager.java

```
@@ -0,0 +1,104 @@
+     * @throws LockError if the locked status of the node could not be determined
+     */
+    boolean isLocked(Node node) throws LockError;
+
+    /**
+     * Check if a subject satisfies all conditions to be locked.
+     * - Not already locked
+     * - Does not have any other conditions meaning that it cannot be locked
+     *
+     * @param node the subject node to check
+     * @return {@code true} if the node can be locked
+     * @throws LockWarning if the node cannot be locked with {@code tryLock}
+     * @throws LockError if it can not be determined if the node can be locked
+     */
+    boolean canLock(Node node) throws LockWarning, LockError;
+
+    /**
+     * Try to lock a subject node.
+     * Will fail if:
+     * - The node is already locked
+     * - There are any other conditions preventing it from being locked
+     *
+     * @param node the subject node to try to lock
+     * @throws LockWarning if the node cannot be locked due to a precondition
+     * @throws LockError if the node cannot be locked for another reason
+     */
+    void tryLock(Node node) throws LockWarning, LockError;
+
+    /**
+     * Try to lock a subject node.
+     * Will fail if the node is already locked
+     *
+     * @param node the subject node to try to lock
+     * @throws LockError if the node could not be locked
+     */
+    void forceLock(Node node) throws LockError;
+
+    /**
+     * Check if a subject node can be unlocked.
+     * Cannot be unlocked if:
+     * - The node is already unlocked
+     * - The node has a locked parent
+     *
+     * @param node the subject node to check
+     * @return {@code true} if the node can be unlocked
+     * @throws LockError if it can not be determined if the node can be unlocked
+     */
+    boolean canUnlock(Node node) throws LockError;
+
+
+    /**
+     * Try to unlock a subject node.
+     * Will fail if:
+     * - The node is already unlocked
```

I'm not sure that unlocking an unlocked node is an error, I see it more like a successful no-op. What do others @data-team-uhn/lfs-devs think?

---

We discussed this in the standup and everyone agreed, will do. Log it as a warning in the backend, silently ignore this case in the front end